### PR TITLE
Add skip_special_tokens in GenerationConfig 

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by AllentDan** [comment]
May search the argument in the tokenzer.py script. There are other three default values need modification.

File: lmdeploy/messages.py [hunk 1]
```diff
@@ -32,6 +32,8 @@ class GenerationConfig:
         bad_words (List[str]): Words that the engine will never generate
         min_new_tokens (int): The minimum numbers of tokens to generate,
             ignoring the number of tokens in the prompt.
+        skip_special_tokens (bool): Whether or not to remove special tokens
```
Comment (RunningLeon) [inline lmdeploy/messages.py:35]: may update docs like here
https://github.com/InternLM/lmdeploy/blob/main/docs/en/inference/pipeline.md#generationconfig

File: lmdeploy/tokenizer.py [hunk 2]
```diff
@@ -121,7 +121,11 @@ def encode(self, s: str, add_bos: bool = True, **kwargs):
         """
         return self.model.Encode(s, add_bos=add_bos, **kwargs)
 
-    def decode(self, t: Sequence[int], offset: Optional[int] = None):
+    def decode(self,
+               t: Sequence[int],
+               offset: Optional[int] = None,
+               *args,
```
Comment (RunningLeon) [inline lmdeploy/tokenizer.py:127]: why add `*args` in here?

**Review by lvhan028 — APPROVED** [review]

File: lmdeploy/serve/turbomind/utils.py [hunk 3]
```diff
@@ -72,22 +72,29 @@ def __init__(self, tritonserver_addr: str):
     def __call__(self, *args, **kwargs):
         return self.infer(*args, **kwargs)
 
-    def infer(self, output_ids: np.ndarray, seqlen: np.ndarray):
+    def infer(self,
+              output_ids: np.ndarray,
+              seqlen: np.ndarray,
+              skip_special_tokens: bool = True):
```
Comment (irexyc) [inline lmdeploy/serve/turbomind/utils.py:78]: skip_special_tokens: bool = True 这里 和 chatbot 传进来的类型对不上。

**Review by irexyc — APPROVED** [review]

**Pull request merged**